### ISSUE TITLE
Outline: Mark current position of editor view

### ISSUE THREAD

**Issue by humdingerb** [Genio-The-Haiku-IDE/Genio#352]
I was wondering...
Would it be useful to mark the symbols in the outline view that currently shown in the editor view? Esp. for long functions, you'd have a reminder in which function you're are right now and what are the functions above/below it.

The outline list could bolden all symbol items that currently visible in the editor view. I suppose the outline list should NOT auto-scroll to have those bolded symbols always visible. This would collide with the user scrolling the outline list 'manually'.

**Comment by humdingerb** [comment]
Just to add, maybe using a tinted background for the marked items would look nicer?

**Comment by nexus6-haiku** [comment]
Just a few thoughts on it. There is no easy way to determine the symbols currently shown in the editor but we may determine the range and do a search in the outline to pick the symbols that completely fall into it. 
As usual this should be done when idle only.

**Comment by jackburton79** [comment]
> I was wondering...
> Would it be useful to mark the symbols in the outline view that currently shown in the editor view? Esp. for long functions, you'd have a reminder in which function you're are right now and what are the functions above/below it.

Yes, it would be very useful. Eclipse (which I used as a "template" for the Outline) selects the symbol in the outline where the caret is in the editor. 
We'd just have to find out a way to do that.

**Comment by Freaxed** [comment]
Implemented in #396

**Comment by Freaxed** [comment]
I see your idea but I'm afraid it can't be implemented simply because the list of symbols may be not following the code order.. For example clangd orders first the attributes then the methods of a class.

I think the feature you describe is the "MiniMap/code-scroller": a zoomed out view of the code that you can quickly navigate.

**Comment by humdingerb** [comment]
Possibly. Maybe as another tab to add tot he right Outline pane someday.
Or some keyboard key that collapses all blocks in the editor view and zoom to 25% while holding. That should compress things nicely and still allows you to orient yourself.

**Comment by jackburton79** [comment]
> Implemented in #396

Nice work!
Can we close this ? The other request seems interesting, but maybe a bit unrealistic for now.
